source/html/renderer/src/host-config.ts

```diff
@@ -69,62 +79,85 @@ function parseColorPaletteDefinition(obj: any): IContainerStyleDefinition {
+    allowCustomStyle: boolean;
 }
 
-function parseAdaptiveCardConfiguration(obj: any): IAdaptiveCardConfig {
+function parseAdaptiveCardConfiguration(obj: any): AdaptiveCardConfig {
     return obj ? {
         allowCustomStyle: obj["allowCustomStyle"]
     } : null;
 }
 
-export interface IImageConfig {
-    size: Enums.Size
+export class ImageConfig {
+    size: Enums.Size;
+
+    toJSON() {
+        return {
+            size: Enums.Size[this.size]
+        }
+    }
 }
 
-function parseImageConfiguration(obj: any): IImageConfig {
-    return obj ? {
+function parseImageConfiguration(obj: any): ImageConfig {
+    return obj ? Object.assign(new ImageConfig(), {
         size: obj["size"]
-    } : null;
+    }) : null;
 }
 
-export interface IImageSetConfig {
+export class ImageSetConfig {
     imageSize: Enums.Size
     maxImageHeight: number
+
+    toJSON() {
+        return {
+            imageSize: Enums.Size[this.imageSize],
+            maxImageHeight: this.maxImageHeight
+        }
+    }
 }
 
-function parseImageSetConfiguration(obj: any): IImageSetConfig {
-    return obj ? {
+function parseImageSetConfiguration(obj: any): ImageSetConfig {
+    return obj ? Object.assign(new ImageSetConfig(), {
         imageSize: obj["imageSize"],
         maxImageHeight: Utils.getValueOrDefault<number>("maxImageHeight", 100)
-    } : null;
+    }) : null;
 }
 
-export interface IFactTextDefinition {
-    size: Enums.TextSize,
-    color: Enums.TextColor,
-    isSubtle: boolean,
-    weight: Enums.TextWeight,
-    wrap: boolean
+export class FactTextDefinition {
+    size: Enums.TextSize;
+    color: Enums.TextColor;
+    isSubtle: boolean;
+    weight: Enums.TextWeight;
+    wrap: boolean;
+
+    toJSON(): any {
+        return {
+            size: Enums.TextSize[this.size],
+            color: Enums.TextColor[this.color],
+            isSubtle: this.isSubtle,
+            weight: Enums.TextWeight[this.weight],
+            warp: this.wrap
+        }
+    }
 }
 
-function parseFactTextDefinition(obj: any): IFactTextDefinition {
-    return obj ? {
+function parseFactTextDefinition(obj: any): FactTextDefinition {
+    return obj ? Object.assign(new FactTextDefinition(), {
         size: Utils.parseHostConfigEnum(Enums.TextSize, obj["size"], Enums.TextSize.Default),
         color: Utils.parseHostConfigEnum(Enums.TextColor, obj["color"], Enums.TextColor.Default),
         isSubtle: obj["isSubtle"],
         weight: Utils.parseHostConfigEnum(Enums.TextWeight, obj["weight"], Enums.TextWeight.Default),
         wrap: obj["wrap"]
-    } : null;
+    }) : null;
 }
 
-export interface IFactTitleDefinition extends IFactTextDefinition {
+export class FactTitleDefinition extends FactTextDefinition {
     maxWidth?: number;
 }
 
-function parseFactTitleDefinition(obj: any): IFactTitleDefinition {
-    var result: IFactTitleDefinition = parseFactTextDefinition(obj);
+function parseFactTitleDefinition(obj: any): FactTitleDefinition {
```

Comment:
Since FactTitleDefinition is now a class, this method should be moved to the class and should be renamed into simply "parse". And it needs to call super.parse(obj) as its first statement.